upstairs/src/client.rs

```diff
@@ -2478,43 +2432,34 @@ async fn proc_stream(
             let fr = FramedRead::new(read, CrucibleDecoder::new());
             let fw = FramedWrite::new(write, CrucibleEncoder::new());
 
-            cmd_loop(rx, stop, tx, fr, fw, log).await
+            cmd_loop(client_id, rx, stop, tx, fr, fw, log).await
         }
         WrappedStream::Https(stream) => {
             let (read, write) = tokio::io::split(stream);
 
             let fr = FramedRead::new(read, CrucibleDecoder::new());
             let fw = FramedWrite::new(write, CrucibleEncoder::new());
 
-            cmd_loop(rx, stop, tx, fr, fw, log).await
+            cmd_loop(client_id, rx, stop, tx, fr, fw, log).await
         }
     }
 }
 
-async fn cmd_loop<R, W>(
-    rx: &mut mpsc::Receiver<Message>,
-    stop: &mut oneshot::Receiver<ClientStopReason>,
-    tx: &mpsc::Sender<ClientResponse>,
+async fn rx_loop<R>(
+    tx: mpsc::Sender<ClientResponse>,
     mut fr: FramedRead<R, crucible_protocol::CrucibleDecoder>,
-    mut fw: FramedWrite<W, crucible_protocol::CrucibleEncoder>,
-    log: &Logger,
+    log: Logger,
```

Comment:
Github is butchering the diff here, sorry about that!

I moved the code from within the `tokio::spawn` into a separate function named `rx_loop`, which now includes the timeout as well as reading from the `FramedRead` (so it's got a `select!` in there)

Reply:
Ah yeah, thanks for the comment otherwise I would have puzzled about this for a while.